### PR TITLE
hw finished, check pls

### PR DESCRIPTION
**Pull request by Watxesnavo**

### CODE REVIEW

File: src/main/java/mate/academy/spring/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final ShoppingCartService cartService;
+    private final UserService userService;
+    private final OrderResponseMapper orderResponseMapper;
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/controller/OrderController.java]: see common mistakes

**Review by aaananas — CHANGES_REQUESTED** [review]
Before you create PR with your homework, you need to go through the checklist under the task and correct all the points described there. The mentor will not check the work until the checklist points are corrected

File: src/main/java/mate/academy/spring/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                           ShoppingCartService cartService,
+                           UserService userService,
+                           DtoResponseMapper<OrderResponseDto, Order> orderResponseMapper) {
+        this.orderService = orderService;
+        this.cartService = cartService;
+        this.userService = userService;
+        this.orderResponseMapper = orderResponseMapper;
+    }
+
+    @PostMapping("/orders/complete")
```
Comment (kozhukhovsky) [inline src/main/java/mate/academy/spring/controller/OrderController.java]:
```suggestion
    @PostMapping("/complete")
```

File: src/main/java/mate/academy/spring/controller/OrderController.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    public void complete(@RequestParam Long userId) {
+        orderService.completeOrder(cartService.getByUser(userService.get(userId)));
+    }
+
+    @GetMapping("/orders")
```
Comment (kozhukhovsky) [inline src/main/java/mate/academy/spring/controller/OrderController.java]:
```suggestion
    @GetMapping
```

File: src/main/java/mate/academy/spring/controller/ShoppingCartController.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+                                  ShoppingCartService shoppingCartService,
+                                  MovieSessionService movieSessionService,
+                                  UserService userService) {
+        this.mapper = mapper;
+        this.shoppingCartService = shoppingCartService;
+        this.movieSessionService = movieSessionService;
+        this.userService = userService;
+    }
+
+    @PutMapping("/shopping-carts/movie-sessions")
```
Comment (kozhukhovsky) [inline src/main/java/mate/academy/spring/controller/ShoppingCartController.java]:
```suggestion
    @PutMapping("/movie-sessions")
```

File: src/main/java/mate/academy/spring/controller/ShoppingCartController.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    public void addMovieSession(@RequestParam Long userId, @RequestParam Long movieSessionId) {
+        shoppingCartService.addSession(movieSessionService.get(movieSessionId),
+                userService.get(userId));
+    }
+
+    @GetMapping("/shopping-carts/by-user")
```
Comment (kozhukhovsky) [inline src/main/java/mate/academy/spring/controller/ShoppingCartController.java]:
```suggestion
    @GetMapping("/by-user")
```

File: src/main/java/mate/academy/spring/controller/UserController.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.response.UserResponseDto;
+import mate.academy.spring.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class UserController {
```
Comment (kozhukhovsky) [inline src/main/java/mate/academy/spring/controller/UserController.java]:
```suggestion
@RestController
@RequestMapping("/users")
public class UserController {
```

File: src/main/java/mate/academy/spring/controller/UserController.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    private final DtoResponseMapper<UserResponseDto, User> userResponseMapper;
+
+    @Autowired
+    public UserController(UserService userService,
+                          DtoResponseMapper<UserResponseDto, User> userResponseMapper) {
+        this.userService = userService;
+        this.userResponseMapper = userResponseMapper;
+    }
+
+    @GetMapping("/users/by-email")
```
Comment (kozhukhovsky) [inline src/main/java/mate/academy/spring/controller/UserController.java]:
```suggestion
    @GetMapping("/by-email")
```

**Review by kozhukhovsky — CHANGES_REQUESTED** [review]
It is generally accepted that through the RequestMapping annotation we have a common path for this controller.
If in further methods GET, PUT and so on, you do not specify an additional path, then it will refer to the address ../yourUrl.
If we want to specify additionally, for example Get("/test)", then it will be ../yourUrl/test.

In your case, for example, it would be ../orders/orders/complete

**Review by kozhukhovsky — APPROVED** [review]

**Pull request closed without merging**